tendrl/node_agent/sds_detection/detect_ceph_sds_details.py

```diff
@@ -0,0 +1,80 @@
+import json
+import logging
+import subprocess
+
+from tendrl.node_agent.auto_detect.sds_detect_plugin import SDSDetectPlugin
+
+LOG = logging.getLogger(__name__)
+
+
+class DetectCephStorageSystem(SDSDetectPlugin):
+    def detect_storage_system(self):
+        ret_val = {}
+
+        # get the gluster version details
+        cmd = subprocess.Popen(
+            "ceph version -f json",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        out, err = cmd.communicate()
+        if err and 'command not found' in err:
+            LOG.error("Error getting gluster version details")
+            return ret_val
+        details = json.loads(out)
+        ret_val['pkg_version'] = details['version'].split()[2]
+        ret_val['pkg_name'] = details['version'].split()[0]
+
+        # get the cluster_id details
+        cmd = subprocess.Popen(
+            "ceph -s -f json",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        out, err = cmd.communicate()
+        if err:
+            LOG.error("Erro getting cluster details")
+            return ret_val
+        details = json.loads(out)
+        ret_val['cluster_id'] = details['fsid']
+        ret_val['cluster_attrs'] = {'fsid': details['fsid'], 'name': 'ceph'}
+
+        # check roles
+        ret_val['roles'] = []
```

Comment:
TAGS will also provide info about the role of the node, Do you think we need it here as well ?